contracts/cw20-stake-external-rewards/src/contract.rs

```diff
@@ -160,10 +189,55 @@ pub fn execute_fund_native(
             let amount = cw_utils::must_pay(&info, &denom).map_err(|_| InvalidFunds {})?;
             execute_fund(deps, env, info.sender, amount)
         }
-        Cw20(_) => Err(InvalidFunds {}),
+        Denom::Cw20(_) => Err(InvalidFunds {}),
     }
 }
 
+pub fn execute_withdraw(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response<Empty>, ContractError> {
+    // only the owner can withdraw
+    let config = CONFIG.load(deps.storage)?;
+    if config.owner != Some(info.sender.clone()) {
+        return Err(UnauthorizedWithdrawal {});
+    };
+
+    // withdrawal is only possible after the reward period
+    let reward_config = REWARD_CONFIG.load(deps.storage)?;
+    if reward_config.period_finish > env.block.height {
+        return Err(RewardPeriodNotFinished {});
+    };
+    let leftover_stake = match &config.reward_token {
```

Comment:
This will break the contract, the reward period may be over but all the rewards might not be claimed, going forward this reward contract will have bugs and a new contract will mostly likely need to be created to distribute future rewards.